### ISSUE TITLE
Temporary variable

### ISSUE THREAD

**Issue by mkstoyanov** [dannys4/FFT-Research-F20#2]
https://github.com/dannys4/FFT-Research-F20/blob/a6a45f6c761b5d84122a1cb1261231d6df839328/FFTE/include/complex.hpp#L41

Do you need a temporary variable here? The more variables the more chances for an error, you can just call:
```
            Complex operator-(const Complex& y) {
                return Complex(  _mm_sub_pd(var, y.var) );
            }
```

**Comment by mkstoyanov** [comment]
The above point counts double here:

https://github.com/dannys4/FFT-Research-F20/blob/a6a45f6c761b5d84122a1cb1261231d6df839328/FFTE/include/complex.hpp#L59

You got it almost right here:
https://github.com/dannys4/FFT-Research-F20/blob/a6a45f6c761b5d84122a1cb1261231d6df839328/FFTE/include/complex.hpp#L72

You just need to remove ret as a temporary.

**Comment by dannys4** [comment]
These should all be fixed, just remnants of when I was returning type `Complex&`. I've addressed this problem in commit [018d7a92f3ce84094cf0840a397d6b6a151a8578](https://github.com/dannys4/FFT-Research-F20/commit/018d7a92f3ce84094cf0840a397d6b6a151a8578)
